server/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardSnapshot.java

```diff
@@ -386,6 +386,19 @@ public BlobStoreIndexShardSnapshot(String snapshot, long indexVersion, List<File
         this.incrementalSize = incrementalSize;
     }
 
+    /**
+     * Creates a new instance has a different name and zero incremental file counts but is identical to this instance in terms of the files
+     * it references.
+     *
+     * @param targetSnapshotName target snapshot name
+     * @param startTime          time the clone operation on the repository was started
+     * @param time               time it took to create the clone
+     */
+    public BlobStoreIndexShardSnapshot asClone(String targetSnapshotName, long startTime, long time) {
```

Comment:
Hmm :) Frankly I like "target" better, leaves less room for ambiguity. With "clone" it's never quite clear if it's the source or the target IMO since by definition "clone" is a somewhat symmetric thing?

Reply:
Sure - I said "maybe" :)